vsphere/datastore_summary_structure.go

```diff
@@ -0,0 +1,72 @@
+package vsphere
+
+import (
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/vmware/govmomi/vim25/types"
+)
+
+// schemaDatastoreSummary returns schema items for resources that
+// need to work with a DatastoreSummary.
+func schemaDatastoreSummary() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		// Note that the following fields are not represented in the schema here:
+		// * Name (more than likely the ID attribute and will be represented in
+		// resource schema)
+		// * Type (redundant attribute as the datastore type will be represented by
+		// the resource)
+		"accessible": &schema.Schema{
+			Type:        schema.TypeBool,
+			Description: "The connectivity status of the datastore. If this is false, some other computed attributes may be out of date.",
+			Computed:    true,
+		},
+		"capacity": &schema.Schema{
+			Type:        schema.TypeInt,
+			Description: "Maximum capacity of the datastore, in bytes.",
+			Computed:    true,
+		},
+		"free_space": &schema.Schema{
+			Type:        schema.TypeInt,
+			Description: "Available space of this datastore, in bytes.",
+			Computed:    true,
+		},
+		"maintenance_mode": &schema.Schema{
+			Type:        schema.TypeString,
+			Description: "The current maintenance mode state of the datastore.",
+			Computed:    true,
+		},
+		"multiple_host_access": &schema.Schema{
+			Type:        schema.TypeBool,
+			Description: "If true, more than one host in the datacenter has been configured with access to the datastore",
+			Computed:    true,
+		},
+		"uncommitted_space": &schema.Schema{
+			Type:        schema.TypeInt,
+			Description: "Total additional storage space, in bytes, potentially used by all virtual machines on this datastore.",
```

Comment:
would this make sense to return MB here rather than bytes? I can't think of a scenario when this would be helpful as Bytes?